cmd/hadolintExecute.go

```diff
@@ -0,0 +1,130 @@
+}
+
+// HadolintClient ..
+// mock generated with: mockery --name hadolintClient --dir cmd --output pkg/hadolint/mocks
+type HadolintClient interface {
+	SetOptions(options piperhttp.ClientOptions)
+	DownloadFile(url, filename string, header http.Header, cookies []*http.Cookie) error
+}
+
+type hadolintRunner interface {
+	RunExecutable(executable string, params ...string) error
+	Stdout(err io.Writer)
+	Stderr(err io.Writer)
+}
+
+type hadolintUtils struct {
+	HadolintPiperFileUtils
+	HadolintClient
+	hadolintRunner
+}
+
+func hadolintExecute(config hadolintExecuteOptions, _ *telemetry.CustomData) {
+	runner := command.Command{
+		ErrorCategoryMapping: map[string][]string{},
+	}
+	// reroute runner output to logging framework
+	runner.Stdout(log.Writer())
+	runner.Stderr(log.Writer())
+
+	utils := hadolintUtils{
+		HadolintPiperFileUtils: &piperutils.Files{},
+		HadolintClient:         &piperhttp.Client{},
+		hadolintRunner:         &runner,
+	}
+
+	if err := runHadolint(config, utils); err != nil {
+		log.Entry().WithError(err).Fatal("Execution failed")
+	}
+}
+
+func runHadolint(config hadolintExecuteOptions, utils hadolintUtils) error {
+	var outputBuffer bytes.Buffer
+	var errorBuffer bytes.Buffer
+	utils.Stdout(&outputBuffer)
+	utils.Stderr(&errorBuffer)
+
+	clientOptions := piperhttp.ClientOptions{
+		TransportTimeout:          20 * time.Second,
+		TransportSkipVerification: true,
+	}
+	if len(config.ConfigurationUsername) > 0 {
+		clientOptions.Username = config.ConfigurationUsername
+		clientOptions.Password = config.ConfigurationPassword
+	}
+	utils.SetOptions(clientOptions)
+
+	options := []string{
+		"--format checkstyle",
+	}
+	// load config file from URL
+	if !hasConfigurationFile(config.ConfigurationFile, utils) && len(config.ConfigurationURL) > 0 {
+		if err := loadConfigurationFile(config.ConfigurationURL, config.ConfigurationFile, utils); err != nil {
+			return errors.Wrap(err, "failed to load configuration file from URL")
+		}
+	}
+	// use config
+	if hasConfigurationFile(config.ConfigurationFile, utils) {
+		options = append(options, fmt.Sprintf("--config %s", config.ConfigurationFile))
+		log.Entry().WithField("file", config.ConfigurationFile).Debug("Using configuration file")
+	} else {
+		log.Entry().Debug("No configuration file found.")
+	}
+	// execute scan command
+	runCommand := fmt.Sprintf("hadolint %s %s", config.DockerFile, strings.Join(options, " "))
+	runCommandTokens := tokenize(runCommand)
+	err := utils.RunExecutable(runCommandTokens[0], runCommandTokens[1:]...)
+
+	//TODO: related to https://github.com/hadolint/hadolint/issues/391
+	// hadolint exists with 1 if there are processing issues but also if there are findings
+	// thus check stdout first if a report was created
+	if output := outputBuffer.String(); len(output) > 0 {
+		log.Entry().WithField("report", output).Debug("Report created")
+		utils.FileWrite(config.ReportFile, []byte(output), 0755)
```

Comment:
shouldn't permissions rather be `0666`?

Reply:
we need to make sure that created files can be deleted after the run outside the container ...

Reply:
done